app/controllers/two_factor_authentication/otp_verification_controller.rb

```diff
@@ -47,6 +45,7 @@ def handle_valid_confirmation_otp
         auth_method: params[:otp_delivery_preference],
       )
       flash[:success] = t('notices.phone_confirmed')
+      user_session.delete(:in_account_creation_flow)
```

Comment:
If a user sets up multiple MFAs during account creation, only the first method would be considered as "in the account creation flow", if we're deleting the session value after each MFA is confirmed. Is that what we want?

I might expect that we only delete the session value once they're truly complete account creation, i.e. they arrive at the partner consent screen or the account dashboard. That's a little tricky currently with how the logic is currently arranged. I'm wondering if, as a first pass, we just don't delete the session value at all? That might risk counting some MFA registrations as "in the account creation flow" if they add from their account dashboard while their initial session is still active, but it doesn't seem too unreasonable to count those as part of "account creation".

Another option is to consider how [`CompletionsController#track_completion_event`](https://github.com/18F/identity-idp/blob/607f6c49c8985066da0b7517c79a65906e51d770/app/controllers/sign_up/completions_controller.rb#L93-L95) is implemented. Not sure the user session deletion makes sense in this method specifically, but it feels like a good precent for "do something after 'completion'".

Reply:
Maybe it'd be fine to just put this in `track_completion_event`, since it's at least related to analytics?

So what I'd imagine is just assigning it in PasswordsController, unsetting it in CompletionsController, and then we don't need to deal with it anywhere else.

Reply:
I dont think we reach the completions controller always though, hence why we need to be wary of where we place it. We only use the completions controller if users are being redirected back/ started from SP.

Reply:
I haven't tested, but I was going by this code, which seemed to suggest the controller (at least at one point) handled the non-SP path:

https://github.com/18F/identity-idp/blob/871eff0b9f96e38e8181fadaf015f91d62051322/app/controllers/sign_up/completions_controller.rb#L65-L68

It would be nice if we could route all traffic through a single location like this to be able to handle these kinds of actions.

I think I'd still be more comfortable with the approach which handles all returned-to-SP users and may have a few stragglers of session value, at least if the alternative is having to try to catch it in all the individual MFA setup controllers. And then maybe we could revisit to make sure the `track_completion_event('account-page')` flow is working correctly.

Reply:
> It would be nice if we could route all traffic through a single location like this to be able to handle these kinds of actions.

More specifically, it seems like we should be able to trust that the event "User registration: complete" is logged correctly when a user completes registration, and if that's not happening, that's a bug we should address separately.

Reply:
> if that's not happening in all cases, that's a bug we should address separately.

I wrote a bug ticket at [LG-11006](https://cm-jira.usa.gov/browse/LG-11006).

Reply:
I think Its not happening currently due to the logic in https://github.com/18F/identity-idp/blob/871eff0b9f96e38e8181fadaf015f91d62051322/app/controllers/application_controller.rb#L227, where #needs_completion_screen will return false unless we have a service provider.

Reply:
We can address here. But sure we can change it, https://github.com/18F/identity-idp/blob/main/app/controllers/concerns/verify_sp_attributes_concern.rb#L2-L18 is the logic for needs completion screen reason.